### PR TITLE
Add memory ballast extension implementation 

### PR DESCRIPTION
**Pull request by bogdandrutu**
Depends on https://github.com/open-telemetry/opentelemetry-collector/pull/1958

### CODE REVIEW

File: extension/ballastextension/testdata/config.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+extensions:
+  memory_ballast:
+  memory_ballast/1:
+    ballast_size_mib: 123
```
Comment (owais) [inline extension/ballastextension/testdata/config.yaml:4]: Could be just `size_mib` as extension name makes the rest obvious.

Comment (bogdandrutu) [reply extension/ballastextension/testdata/config.yaml:4]: nice, will do that.

Comment (bogdandrutu) [reply extension/ballastextension/testdata/config.yaml]: Done.

**Comment by bogdandrutu** [comment]
@tigrannajaryan ping

**Pull request closed without merging**